util/ci/indent_tab_preprocess.py

```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/python3
+
+import sys
+import re
+
+# preprocess files for indent check
+
+# Code processing:
+#
+# Tabs are keeped
+# Spaces after tabs is ignored if indent is not smaller
+#
+# Example:
+#
+# <tab>a = 6;
+# <spc>b = 5;
+# <tab><spc>if (
+# <tab><tab><tab><spc><spc>) {
+# <tab><tab>c = 7;
+# <tab><spc>}
+#
+# Result:
+#
+# <tab>a = 6;
+# <spc>b = 5;
+# <tab>if (
+# <tab><tab><tab>) {
+# <tab><tab>c = 7;
+# <tab><spc>}
+
+# Multiline comments processing:
+#
+# It is expected that tabs indent from comment begining is
+# used as indent on begining of every commented line.
+#
+# Example:
+#
+# <tab>/*
+# <spc><spc>A
+# <tab><tab>B
+# <tab><spc>C
+# D
+# <tab><spc>*/
+# <tab><tab>/*
+# <spc><tab><spc>R
+#	<tab><spc><tab>*/
+#
+# Result:
+#
+# <tab>/*
+# <spc>A
+# <tab>B
+# <tab>C
+# D
+# <tab>*/
+# <tab><tab>/*
+# <spc><tab>*R
+#	<tab><spc>*/
+
+
+def main():
+	if len(sys.argv) != 5:
+		#print("Bad arguments")
+		#print(sys.argv)
+		exit()
+	# multiline comment begining and end from command line
+	mlc_begin = sys.argv[1]
+	mlc_end = sys.argv[2]
+	mlc_end_len = int(sys.argv[3])
+	file = open(sys.argv[4], "r")
+
+	re_mlc_begin = "^.*{}".format(mlc_begin)
+	re_mlc_begin_end = "^.*{}.*{}".format(mlc_begin, mlc_end)
+	re_mlc_end = "^.*{}".format(mlc_end)
```

Comment:
Wouldn't simple non-pattern string search be easier?

Reply:
For begin and end, probably yes. For the `begin_end `combination, I do not thing so.

Reply:
You'd first search for the begin, then for the end, with the offset. But whatever.